### ISSUE TITLE
Large Slider displays text in precentages

### ISSUE THREAD

**Issue by cmccullough2** [cmv/cmv-app#581]
What are the correct parameters for the slider? 
When I set the sliderLabels to labels the site will not render properly 
when I set the sliderLabels to true the labels display in percentages. 
![scalebarper](https://cloud.githubusercontent.com/assets/3169799/17520361/16416312-5e15-11e6-8470-5f339532a158.PNG)
This is what I want to see, https://developers.arcgis.com/javascript/3/samples/mapconfig_customlabels/

**Comment by roemhildtg** [comment]
What have you tried setting? Can you post your config for the widget so someone can reproduce the issue?

**Comment by cmccullough2** [comment]
Unfortunately I cannot.  This is my configurations in the viewer.js.
 mapOptions: {
            basemap: 'streets',
            center: [-96.59179687497497, 39.09596293629694],
            zoom: 5,
            sliderStyle: 'large',
            sliderLabels: labels

**Comment by tmcgee** [comment]
@cmccullough2 Did you create the labels first like in this [sample](https://developers.arcgis.com/javascript/3/jssamples/mapconfig_customlabels.html)?

**Comment by cmccullough2** [comment]
I did not.  Where should I make these edits?

**Comment by tmcgee** [comment]
It would need to be in viewer.js before the mapOptions. If you know what the labels should be, you can just hard-code them in an array. If you need to retrieve them from the LODs of a layer use the code in the sample. Something like this at the top of viewer.js before the return statement: 

``` javascript
var layer = new ArcGISTiledMapServiceLayer("https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer");
// once the layer loads, use its tile levels to label the map's slider
layer.on("load", function() {
    // use layer's scales to display as slider labels
    var labels = [];
    var lods = layer.tileInfo.lods;
    for ( var i = 0, il = lods.length; i < il; i++ ) {
        if ( i % 2 ) {
            labels.push(number.format(lods[i].scale.toFixed()));
        }    
    }
}
```

To use this code as written, you'll need to include `esri/layers/ArcGISTiledMapServiceLayer` and `dojo/number` to the list of modules with the appropriate module names passed to the function.

**Comment by cmccullough2** [comment]
I have tried a few options, but not winning. The modules have been added to the viewer.js  Sorry I am stumped and still trying.  
![scalebarper3](https://cloud.githubusercontent.com/assets/3169799/17526233/0898a880-5e2c-11e6-878f-db249e872a0c.PNG)
This is my viewer.js file configurations:
![scalebarper2](https://cloud.githubusercontent.com/assets/3169799/17526187/d9228594-5e2b-11e6-9612-3337df41d096.PNG)
And my error message:
![Uploading ScaleBarPer3.PNG…]()

**Comment by cmccullough2** [comment]
my images inverted above.

**Comment by tmcgee** [comment]
@cmccullough2 you cannot put the code within the JSON. That is the cause of your error. That code block needs to be above the `return` statement as I mentioned in my comment

**Comment by cmccullough2** [comment]
I am receiving error message 'labels' is undefined.  The js file also needs to close the var layer function with a );.  
var layer = new ArcGISTiledMapServiceLayer("https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer");
				// once the layer loads, use its tile levels to label the map's slider
				layer.on("load", function() {
				    // use layer's scales to display as slider labels
				    var labels = [];
				    var lods = layer.tileInfo.lods;
				    for ( var i = 0, il = lods.length; i < il; i++ ) {
				        if ( i % 2 ) {
				            labels.push(number.format(lods[i].scale.toFixed()));
				        }    
				    }
				});
    return {
        // used for debugging your app
        isDebug: true,

**Comment by cmccullough2** [comment]
It is baffling to me why the labels are showing %'s instead of numbers.

**Comment by tmcgee** [comment]
Right. You need a slight change from the example. Move the line `var labels = [];` above the line starting with `var layer =`

**Comment by cmccullough2** [comment]
Ah!  I will try this.

**Comment by cmccullough2** [comment]
I tried it.  It did not work.  I have to move onto something else. I may add the LODs in the array.

**Comment by tmcgee** [comment]
@cmccullough2 2 things:

1. You will have to switch the the full version of the Esri JavaScript API. By default, CMV loads the compact version of the API,

2. It would be much easier to hard-code the labels in your `labels`array.  As the code is written now, the labels array is empty in the mapOptions because it is populated with values only after that layer is loaded which is some amount of time (milliseconds) after the rest of the code in viewer.js executes.

I have confirmed that the above works as I would expect.

**Comment by cmccullough2** [comment]
I was able to display the labels as suggested.  I will need to decrease the font, but this will work for now with the code below. I am using the full api reference. 
![scalebarper4](https://cloud.githubusercontent.com/assets/3169799/17555752/5a6005dc-5ed6-11e6-9ae6-7cfb7841b1c9.PNG)

var labels = [];
				var layer = new ArcGISTiledMapServiceLayer("https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer");
				// once the layer loads, use its tile levels to label the map's slider
				layer.on("load", function() {
				    // use layer's scales to display as slider labels
				   
				    var lods = layer.tileInfo.lods;
				    for ( var i = 0, il = lods.length; i < il; i++ ) {
				        if ( i % 2 ) {
				            labels.push(number.format(lods[i].scale.toFixed()));
				        }    
				    }
				});
    return {
        // used for debugging your app
        isDebug: true,

        //default mapClick mode, mapClickMode lets widgets know what mode the map is in to avoid multipult map click actions from taking place (ie identify while drawing).
        defaultMapClickMode: 'identify',
        // map options, passed to map constructor. see: https://developers.arcgis.com/javascript/jsapi/map-amd.html#map1
        mapOptions: {
            basemap: 'streets',
            logo: false,
            center: [-96.59179687497497, 39.09596293629694], 
            zoom: 5,
            sliderStyle: 'large',
            sliderLabels: [147914382, 36978595, 9244649, 2311162, 577790, 144447, 36111, 9027, 2256, 564]
        },

**Comment by tmcgee** [comment]
@cmccullough2 Just a quick comment to point out that with hard-coded labels, you no longer need any of the code related to retrieving the labels fro the layer and that can be removed.

**Comment by cmccullough2** [comment]
@tmcgee I was wondering if I still needed the snippet of code.  Thanks for your help.

**Comment by mayur9785** [comment]
Hi,
I added slider with labels but its not proper what settings do i need to do so it should display proper on top-left and below slider it display homebutton and locate button

![slider](https://cloud.githubusercontent.com/assets/6058868/18043655/ef2d68b2-6dd1-11e6-810f-0a694191ac24.png)

I tried to change this css 
```
.HomeButton 
{
   margin-top:120px;
}
```
then it coming down but then unable drag & click sliderbar
